modules/mono/csharp_script.cpp

```diff
@@ -3498,8 +3498,18 @@ void CSharpScript::get_script_property_list(List<PropertyInfo> *p_list) const {
 }
 
 int CSharpScript::get_member_line(const StringName &p_member) const {
-	// TODO omnisharp
-	return -1;
+	Vector<String> source_code_line = get_source_code().split("\n");
+	int p_line = -1;
+	//int p_col = 0;
+	for (int i = 0; i < source_code_line.size(); i = i + 1) {
+		int tmp_col = source_code_line[i].find("void " + p_member);
```

Comment:
If I understand correctly this would only work for methods which return `void`? That seems quite limited then.

It can also find the wrong occurrence if you have several methods that have part of their name shared:

```csharp
public void take_damage_player() {};
public void take_damage_enemy() {};
public void take_damage() {};
```

If `p_member` is `take_damage()` it would focus `take_damage_player()` as the first occurrence found.

Reply:
按照GIF中的演示,信号都是'void',使用是没问题的.
我应该使用正则表达式,当时为了偷懒,我很抱歉,现在已经改进好了.

According to the demo in the GIF, the signals are all 'void', which is fine to use.
I should have used regular expressions, I'm sorry for being lazy at the time, but I've improved it now.

Reply:
Not only what @akien-mga mentioned, but also the source code in the external editor may not be the same as the one Godot has loaded or the one on disk (e.g.: when there are unsaved changes).
IMO implementing this properly requires sending a "go to method" request and delegating the job to the editor/IDE. Maybe with a fallback to this PRs solution if the former is not supported.